pkg/abcupdater/version_test.go

```diff
@@ -206,3 +207,44 @@ To disable notifications for this new version, set SAMPLE_APP_1_IGNORE_VERSIONS=
 		})
 	}
 }
+
+// Note: These tests rely on timing and could be flaky if breakpoints are used.
+func Test_asyncFunctionCall(t *testing.T) {
+	t.Parallel()
+	cases := []struct {
+		name  string
+		input func() (string, error)
+		want  string
+	}{
+		{
+			name:  "happy_path",
+			input: func() (string, error) { return "done", nil },
+			want:  "done\n",
+		},
+		{
+			name:  "error_path_still_returns",
+			input: func() (string, error) { return "", fmt.Errorf("failed") },
+			want:  "",
+		},
+		{
+			// This test will take a long time due to forcing timeout.
+			name:  "timeout_gets_applied",
+			input: func() (string, error) { time.Sleep(9999 * time.Hour); return "should_not_execute", nil },
+			want:  "",
+		},
+		// TODO: would like a clean way to test the canceled context case, not sure
```

Comment:
You can always define a new test function, like `func TestAsyncFunctionCall_ContextCanceled(t *testing.T { ... ` . It's not required to make everything a subtest under one top-level function.